aderyn_core/src/detect/experimental/parent_chain.rs

```diff
@@ -26,20 +26,25 @@ In ParentChainContract.sol, there is only 1 assignment done. The goal is to capt
 impl IssueDetector for ParentChainDemonstrator {
     fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
         for assignment in context.assignments() {
-            println!("0 {}", assignment);
             capture!(self, context, assignment);
 
             if let Some(parent_chain) = assignment.parent_chain(context) {
-                if let ASTNode::Block(_) = parent_chain[0] {
-                    capture!(self, context, parent_chain[0]);
-                }
-                if let ASTNode::ForStatement(_) = parent_chain[1] {
+                if let ASTNode::Block(_) = parent_chain[1] {
                     capture!(self, context, parent_chain[1]);
                 }
-                if let ASTNode::Block(block) = parent_chain[2] {
+                if let ASTNode::ForStatement(_) = parent_chain[2] {
+                    capture!(self, context, parent_chain[2]);
+                }
+                if let ASTNode::Block(block) = parent_chain[3] {
                     capture!(self, context, block);
                 }
             }
+
+            if let Some(mut parent_chain) = assignment.parent_chain(context) {
+                let sorted_chain = parent_chain.sort_by_line_nos(context).unwrap();
+                parent_chain.reverse();
+                assert_eq!(sorted_chain, parent_chain);
+            }
```

Comment:
Here you can see that the `parent_chain` when reversed becomes equal to the sorted version of itself. !!

Reply:
(Example)